### PR TITLE
Remove RelationColumn

### PR DESCRIPTION
**Pull request by mfussenegger**
The RelationColumn was used in the case of a join, to plan
query-then-fetch. It was kind of a Field/InputColumn hybrid with a
pre-calculated index.
This pre-calculated index allowed fast index-lookups, but had the
disadvantage that any further query rewrites became difficult as all
indices had to be updated correctly.
In addition the join-planner components had to deal with two different
kind of symbols: Field and RelationColumn. This required a lot of symbol
mapping logic.

As a side effect of this change the "requiredForQuery" logic is also a
bit more optimized.

Before in a query like

    select coalesce(t1.name, ''), coalesce(t2.name, '') from t1, t2
    order by 1, 2

We'd have both `coalesce(t1.name, '')` and `t1.name` as
`requiredForQuery`, now only `t1.name` is required.

This is important because in outer-join cases any functions which can
cause null-values to become non-null have to be applied *after* the
join.

### CODE REVIEW

File: sql/src/main/java/io/crate/planner/consumer/NestedLoopConsumer.java [hunk 1]
```diff
@@ -247,8 +225,8 @@ public Plan visitTwoTableJoin(TwoTableJoin statement, ConsumerContext context) {
             if (joinCondition != null) {
                 joinCondition = InputColumns.create(joinCondition, nlOutputs);
                 assert joinCondition instanceof Function : "Only function symbols are valid join conditions";
-                boolean hasRelCol = SymbolVisitors.any(s -> s instanceof RelationColumn, joinCondition);
-                assert  !hasRelCol : "RelationColumns are not valid join condition arguments";
+                boolean hasColumn = SymbolVisitors.any(Symbols.IS_COLUMN, joinCondition);
```
Comment (matriv) [inline sql/src/main/java/io/crate/planner/consumer/NestedLoopConsumer.java]: Can collapse into one line assertion

File: sql/src/main/java/io/crate/analyze/relations/DocTableRelation.java [hunk 2]
```diff
@@ -90,9 +87,6 @@ public Field getField(Path path) {
     @Override
     public Field getField(Path path, Operation operation) throws UnsupportedOperationException, ColumnUnknownException {
         ColumnIdent ci = toColumnIdent(path);
-        if (HIDDEN_COLUMNS.contains(ci)) {
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/relations/DocTableRelation.java]: how is this related to your commit?

Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/relations/DocTableRelation.java]: It's necessary to be able to resolve the _fetchId  here: https://github.com/crate/crate/pull/5493/files#diff-630bac56f4217582a384868fccb29c64R77
